packages/flutter_tools/lib/src/flx.dart

```diff
@@ -36,7 +36,7 @@ Future<int> createSnapshot({
   assert(mainPath != null);
   assert(snapshotPath != null);
   assert(packages != null);
-  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot);
+  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot, null, BuildMode.debug);
```

Comment:
To build a script snapshot (as is done here) any gen_snapshot binary should be sufficient. In non-local-engine mode, we return gen_snapshot in profile mode for example. Specifying debug mode here feels wrong.